app/src/main/java/com/nevilleantony/prototype/adapters/CompletedListAdapter.java

```diff
@@ -0,0 +1,62 @@
+package com.nevilleantony.prototype.adapters;
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.nevilleantony.prototype.R;
+import com.nevilleantony.prototype.downloadmanager.DownloadService;
+
+import java.io.File;
+import java.util.List;
+
+public class CompletedListAdapter extends RecyclerView.Adapter<CompletedListAdapter.RecyclerViewHolder> {
+
+    private final List<File> downloads;
+    private final Intent downloadIntent;
+    private final Context context;
+
+    public CompletedListAdapter(Context context, List<File> completedList) {
+        this.downloads = completedList;
+        downloadIntent = new Intent(context, DownloadService.class);
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view = inflater.inflate(R.layout.fragment_completed_list_row, parent, false);
+        return new RecyclerViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerViewHolder holder, int position) {
+        File fileDownload = downloads.get(position);
+        holder.downloadsName.setText(fileDownload.getName());
```

Comment:
Here you also need to set an onClick listener on the share button, which creates an intent for ShareActivity.class. You should also add the absolute path of the file as an extra in the intent and call startActivity with it.